Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -413,8 +421,11 @@ final class PurchasesOrchestrator {
 
         let addPayment: Bool = self.addPurchaseCompletedCallback(
             productIdentifier: productIdentifier,
-            completion: { transaction, customerInfo, error, cancelled in
+            completion: { [weak self] transaction, customerInfo, error, cancelled in
+                guard let self = self else { return }
```

Comment:
without this, we were leaking `PurchasesOrchestator` and `DeviceCache` 

cc: @tonidero

Reply:
Hmm interesting... I'm not sure I understand the reason...  I think this should be mostly safe, except maybe we're not ensuring the completion blocks to be called if changing the purchases orchestrator instance for any reason... Which should normally be ok I think.

In any case, would it be possible to do this inside the `trackPurchaseEventIfNeeded` if that was the reason for the leak to avoid changing any other code?

Reply:
well the reason is that the completion block holds a reference to `self`, and we are adding the completion block to the `purchaseCompletedCallbacks`, which I think are not getting triggered when testing, so even if I add it to `trackPurchaseEventIfNeeded`, we still need to call `self. trackPurchaseEventIfNeeded`, and the completion would still be holding the reference to `self`. Am I misunderstanding something?

Reply:
Ohh ok right. Make sense. Thanks for the explanation!